debezium-core/src/main/java/io/debezium/pipeline/ChangeEventSourceCoordinator.java

```diff
@@ -82,51 +82,43 @@ public ChangeEventSourceCoordinator(TaskOffsetContext<P, O> previousOffsetContex
 
     public synchronized <T extends CdcSourceTaskContext> void start(T taskContext, ChangeEventQueueMetrics changeEventQueueMetrics,
                                                                     EventMetadataProvider metadataProvider) {
-        O tempOffset = null;
-        P tempPartition = null;
-
-        // TODO: iterate over all entries instead of breaking
-        for (Map.Entry<P, O> entry : previousOffsetContext.getOffsets().entrySet()) {
-            tempPartition = entry.getKey();
-            tempOffset = entry.getValue();
-            break;
-        }
-
         this.snapshotMetrics = changeEventSourceMetricsFactory.getSnapshotMetrics(taskContext, changeEventQueueMetrics, metadataProvider);
         this.streamingMetrics = changeEventSourceMetricsFactory.getStreamingMetrics(taskContext, changeEventQueueMetrics, metadataProvider);
         running = true;
 
-        P partition = tempPartition;
-        O previousOffset = tempOffset;
-
         // run the snapshot source on a separate thread so start() won't block
         executor.submit(() -> {
             try {
-                snapshotMetrics.register(LOGGER);
-                streamingMetrics.register(LOGGER);
-                LOGGER.info("Metrics registered");
-
-                ChangeEventSourceContext context = new ChangeEventSourceContextImpl();
-                LOGGER.info("Context created");
-
-                SnapshotChangeEventSource<P, O> snapshotSource = changeEventSourceFactory.getSnapshotChangeEventSource(snapshotMetrics);
-                CatchUpStreamingResult catchUpStreamingResult = executeCatchUpStreaming(context, previousOffset, snapshotSource);
-                if (catchUpStreamingResult.performedCatchUpStreaming) {
-                    streamingConnected(false);
-                    commitOffsetLock.lock();
-                    streamingSource = null;
-                    commitOffsetLock.unlock();
-                }
-                eventDispatcher.setEventListener(snapshotMetrics);
-                SnapshotResult<O> snapshotResult = snapshotSource.execute(context, partition, previousOffset);
-                LOGGER.info("Snapshot ended with {}", snapshotResult);
-
-                if (snapshotResult.getStatus() == SnapshotResultStatus.COMPLETED || schema.tableInformationComplete()) {
-                    schema.assureNonEmptySchema();
-                }
-
-                if (running && snapshotResult.isCompletedOrSkipped()) {
-                    streamEvents(context, partition, snapshotResult.getOffset());
+                for (Map.Entry<P, O> entry : previousOffsetContext.getOffsets().entrySet()) {
```

Comment:
I've put the loop into the thread but it won't work in case of multiple not `INITTIAL_ONLY` snapshots. Streaming phase will block the next snapshot.